coverage_config_x86_64.json

```diff
@@ -1,5 +1,5 @@
 {
-  "coverage_score": 46.8,
+  "coverage_score": 46.1,
   "exclude_path": "",
```

Comment:
it's a shame coverage keeps going down.

Reply:
It goes up as well sometimes :)

Reply:
It looks a bit like a trend at this point with the coverage going down. What do you think about adding some tests first, and only after merging these PRs? This would help with providing the right infrastructure for adding tests for added functionality as the functionality gets merged. This should also bring more confidence that the functionality is correct.

Reply:
This pull request do contain new test for the newly added feature and this pull request in particular needs to be merged now since it doesn't work with the new kernel otherwise.

On testing front I do agree that we need to do more. As I understood until now the coverage would increase only once we add tests for the code that  isn't getting tested at all, mostly vhu_i2c.rs i.e..

I think @mathieupoirier also had a look at that stuff. It wasn't really straight forward to implement and would likely need modification to other crates as well to provide mock implementations. And that's why it had been lagging since so long, I tried to look into it multiple times but wasn't able to figure out the right way of doing it.

Having said that, I think we can add more tests for the rest of the code as the layout is much better to add proper tests. I will work on that.

Reply:
Providing an adequate test infrastructure for the vring manipulation in vhu_i2c.rs and vhu_rng.rs is something that needs to be done.  As stated in my conversation with @andreeaflorescu [here](https://github.com/rust-vmm/vhost-device/pull/29), I may have found a way forward.  I will invest time on implementing that solution next week.